### PR TITLE
InsightConnect plugin hub updates by Gursimran

### PR DESCRIPTION
**Pull request by gusingh-r7**
## Proposed Changes

Describe the proposed changes:

  - Awk, AWS workspaces, AD/LDAP hub update

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `./run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `./run -T tests/example.json -d -j`
    * All actions/triggers shortcut `./run -T all -d -j` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `./run -R tests/example.json -d -j`
    * All actions/triggers shortcut `./run -R all -d -j` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `./run -A -R all -T all`. For single action validation: `./run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`./run ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

File: aws_workspaces/plugin.spec.yaml [hunk 1]
```diff
@@ -7,13 +7,17 @@ vendor: rapid7
 support: community
 status: []
 resources:
-  source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/aws_workspaces
+  source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/active_directory_ldap
```
Comment (jherlitz-r7) [inline aws_workspaces/plugin.spec.yaml]:
```suggestion
  source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/aws_workspaces
```

File: advanced_regex/help.md [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
-* Feature 3
+* Data extraction
+* Search and replace text
+* Split text
 
 # Requirements
-
-* Example: Requires an API Key from the product
-* Example: API must be enabled on the Settings page in the product
+This plugin does not contain any requirements.
```
Comment (jschipp-r7) [inline advanced_regex/help.md]:
```suggestion
_This plugin does not contain any requirements._
```

File: awk/help.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Description
 
-[Awk](https://www.gnu.org/software/gawk/manual/gawk.html) is a pattern scanning and processing language. This plugin uses GNU Awk to manipulate input data.
+The Awk InsightConnect plugin manipulate input data using GNU [Awk](https://www.gnu.org/software/gawk/manual/gawk.html)
```
Comment (jschipp-r7) [inline awk/help.md]: manipulates

File: awk/help.md [hunk 4]
```diff
@@ -15,14 +17,11 @@ Second column contents: birds
 
 # Key Features
 
-* Feature 1
-* Feature 2
-* Feature 3
+* Manipulate string
```
Comment (jschipp-r7) [inline awk/help.md]: Search and replace

File: awk/help.md [hunk 5]
```diff
@@ -15,14 +17,11 @@ Second column contents: birds
...
+* Manipulate file
 
 # Requirements
-
-* Example: Requires an API Key from the product
-* Example: API must be enabled on the Settings page in the product
+This plugin does not contain any requirements.
```
Comment (jschipp-r7) [inline awk/help.md]: Use this style

```suggestion
_This plugin does not contain any requirements._
```

File: awk/plugin.spec.yaml [hunk 6]
```diff
@@ -3,20 +3,21 @@ extension: plugin
 product: ["insightconnect"]
 name: awk
 title: Awk
-description: Manipulate text with the GNU awk programming language
+description: This plugin manipulate input string or file with the GNU awk programming language
```
Comment (jschipp-r7) [inline awk/plugin.spec.yaml]: manipulates

File: awk/plugin.spec.yaml [hunk 7]
```diff
@@ -3,20 +3,21 @@ extension: plugin
...
 version: 1.2.0
 vendor: rapid7
 support: community
 status: []
 resources:
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/awk
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://www.gnu.org/software/gawk/manual/gawk.html
```
Comment (jschipp-r7) [inline awk/plugin.spec.yaml]: vendor, not documentation

File: advanced_regex/plugin.spec.yaml [hunk 8]
```diff
@@ -1,16 +1,22 @@
...
-description: Perform advanced regex operations on a string using Python regex
+description: This plugin perform advanced regex operations on a string using Python regex
 version: 1.0.0
 support: community
 vendor: rapid7
 status: []
+resources:
+  source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/advanced_regex
+  license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://docs.python.org/3/library/re.html
```
Comment (jschipp-r7) [inline advanced_regex/plugin.spec.yaml]: vendor URL, fix for all

File: advanced_regex/help.md (hunk 2 above)
Comment (jschipp-r7) [reply advanced_regex/help.md]: Fix this style for remaining

File: basename/help.md [hunk 9]
```diff
@@ -14,14 +15,10 @@ This plugin is used to get the last item of a file path or URL. Examples, using
 
 # Key Features
 
-* Feature 1
-* Feature 2
-* Feature 3
+* Delete input's prefix ending with last slash '/' character
```
Comment (jschipp-r7) [inline basename/help.md]: Obtain basename of a file path or URL

File: diff/plugin.spec.yaml [hunk 10]
```diff
@@ -11,11 +11,12 @@ version: 1.0.1
 resources:
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/diff
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://docs.python.org/3/library/difflib.html
```
Comment (jherlitz-r7) [inline diff/plugin.spec.yaml]:
```suggestion
  vendor_url: https://www.rapid7.com/
```

File: dirname/help.md [hunk 11]
```diff
@@ -1,7 +1,7 @@
 # Description
 
-The `dirname` utility deletes any suffix beginning with the slash `/` character
-This plugin is used to get the directory name of a file path or protocol and domain of a URL. Examples, using Python
+The `dirname` InsightConnect plugin is used to get the directory name of a file path or protocol and domain of a URL. This deletes any suffix beginning with the slash `/` character.
+ Examples, using Python:
```
Comment (jschipp-r7) [inline dirname/help.md]: Should be on a newline to pass `mdl` linting

```

Examples...:

...
```

File: freegeoip/plugin.spec.yaml [hunk 12]
```diff
@@ -3,20 +3,21 @@ extension: plugin
...
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/freegeoip
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://freegeoip.net
 tags:
 - geoip
 - ip
 - location
 hub_tags:
-  use_cases: []
+  use_cases: [log_search]
```
Comment (jschipp-r7) [inline freegeoip/plugin.spec.yaml]: This is threat_intel

File: freeipa/help.md [hunk 13]
```diff
@@ -1,19 +1,17 @@
...
+The `freeipa` InsightConnect plugin retrieve user details, status and delete user from [FreeIPA](https://www.freeipa.org/) which is an integrated Identity and Authentication solution for Linux/UNIX networked environments.
 This plugin runs commands on a FreeIPA server using this [freeipa](https://github.com/nordnet/python-freeipa-json) python library.
 The FreeIPA API is viewable on the FreeIPA server web interface.
 
 # Key Features
 
-* Feature 1
-* Feature 2
-* Feature 3
+* Search user and retrieve user details
```
Comment (jschipp-r7) [inline freeipa/help.md]: User management

File: freeipa/help.md [hunk 14]
```diff
@@ -1,19 +1,17 @@
...
+* Delete user
 
 # Requirements
 
-* Example: Requires an API Key from the product
-* Example: API must be enabled on the Settings page in the product
+* Username and Password to logon to [freeipa](https://www.freeipa.org/) server.
```
Comment (jschipp-r7) [inline freeipa/help.md]: lowercase password

File: active_directory_ldap/plugin.spec.yaml [hunk 15]
```diff
@@ -3,19 +3,20 @@ extension: plugin
...
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/active_directory_ldap
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://docs.microsoft.com/en-us/windows-server/identity/ad-ds/active-directory-domain-services
 tags:
 - ldap
 - active directory
 hub_tags:
-  use_cases: []
+  use_cases: [user_management, credential_management, application_security]
   keywords: [ldap, active directory]
```
Comment (mhofert-r7) [inline active_directory_ldap/plugin.spec.yaml]: add `microsft` as a keyword

File: freegeoip/help.md [hunk 16]
```diff
@@ -1,17 +1,14 @@
...
 
-[FreeGeoIP](https://freegeoip.net/) is a free service for Geo IP lookups.
+The `freegeoip` InsightConnect plugin lookup GeoIP information for specified host using [FreeGeoIP](https://freegeoip.net/) which is a free service for Geo IP lookups.
 
 # Key Features
 
-* Feature 1
-* Feature 2
-* Feature 3
+* GeoIP lookup
```
Comment (jschipp-r7) [inline freegeoip/help.md]: Obtain GeoIP information for an IP address

File: advanced_regex/help.md (hunk 2 above)
Comment (mhofert-r7) [inline advanced_regex/help.md]: `_This plugin does not contain any requirements._` add a _ to both sides

File: freeipa/plugin.spec.yaml [hunk 17]
```diff
@@ -3,20 +3,20 @@ extension: plugin
 product: ["insightconnect"]
 name: freeipa
 title: FreeIPA
-description: FreeIPA is an integrated Identity and Authentication solution for Linux/UNIX
-  networked environments
+description: This plugin retrieve user details, status and delete user from FreeIPA server
```
Comment (jherlitz-r7) [inline freeipa/plugin.spec.yaml]:
```suggestion
description: This plugin can retrieve user details and statuses and delete users from FreeIPA servers
```

File: aws_workspaces/plugin.spec.yaml [hunk 18]
```diff
@@ -9,11 +9,15 @@ status: []
...
+  vendor_url: https://aws.amazon.com/workspaces
+description: This plugin utilizes Amazon WorkSpaces product to create workspaces. Amazon WorkSpaces is a fully managed
+  desktop computing service in the cloud that allows its customers to provide cloud-based desktops to their end-users
+version: 1.0.0
 tags:
 - aws
 - workspaces
 hub_tags:
-  use_cases: []
+  use_cases: [asset_inventory]
```
Comment (jschipp-r7) [inline aws_workspaces/plugin.spec.yaml]: application_management

File: ftp/plugin.spec.yaml [hunk 19]
```diff
@@ -6,16 +6,17 @@ title: FTP
 vendor: rapid7
 support: community
 status: []
-description: Upload and download files using the File Transfer Protocol
+description: This plugin upload, download and delete files between a client and server on a computer network using the File Transfer Protocol
```
Comment (jherlitz-r7) [inline ftp/plugin.spec.yaml]:
```suggestion
description: This plugin can upload, download and delete files between a client and server on a computer network using the File Transfer Protocol
```

File: ftp/plugin.spec.yaml [hunk 20]
```diff
@@ -6,16 +6,17 @@ title: FTP
...
 version: 2.0.0
 resources:
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/ftp
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url: https://docs.python.org/3/library/ftplib.html
```
Comment (jherlitz-r7) [inline ftp/plugin.spec.yaml]:
```suggestion
  vendor_url: https://www.rapid7.com/
```

File: awk/help.md (hunk 5 above)
Comment (mhofert-r7) [inline awk/help.md]: see above

File: base64/help.md [hunk 21]
```diff
@@ -5,14 +5,11 @@ This plugin utilizes allows data to be encoded or decoded using the standard Bas
...
-* Feature 2
-* Feature 3
+* Encode text
+* Decode text
 
 # Requirements
-
-* Example: Requires an API Key from the product
-* Example: API must be enabled on the Settings page in the product
+This plugin does not contain any requirements.
```
Comment (mhofert-r7) [inline base64/help.md]: see above

File: base64/plugin.spec.yaml [hunk 22]
```diff
@@ -11,13 +11,14 @@ status: []
 resources:
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/base64
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url:
```
Comment (mhofert-r7) [inline base64/plugin.spec.yaml]: https://www.rapid7.com

File: basename/help.md [hunk 23]
```diff
@@ -14,14 +15,10 @@ This plugin is used to get the last item of a file path or URL. Examples, using
...
 
 # Requirements
-
-* Example: Requires an API Key from the product
-* Example: API must be enabled on the Settings page in the product
+This plugin does not contain any requirements.
```
Comment (mhofert-r7) [inline basename/help.md]: see above

File: basename/plugin.spec.yaml [hunk 24]
```diff
@@ -3,19 +3,20 @@ extension: plugin
...
-description: Basename a path
+description: This plugin is used to get the last item of a file path or URL using Python's basename utility.
 version: 1.0.0
 vendor: rapid7
 support: community
 status: []
 resources:
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/basename
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url:
```
Comment (mhofert-r7) [inline basename/plugin.spec.yaml]: https://www.rapid7.com

File: datetime/help.md [hunk 25]
```diff
@@ -1,17 +1,16 @@
...
-* Feature 3
+* Convert a datetime to an Epoch and vice versa
+* Convert a datetime to specified format
+* Determine the elapsed time between two dates
 
 # Requirements
 
-* Example: Requires an API Key from the product
-* Example: API must be enabled on the Settings page in the product
+This plugin does not contain any requirements.
```
Comment (mhofert-r7) [inline datetime/help.md]: see above

File: diff/help.md [hunk 26]
```diff
@@ -1,18 +1,15 @@
...
-* Feature 1
-* Feature 2
-* Feature 3
+* Get differences between strings.
 
 # Requirements
 
-* Example: Requires an API Key from the product
-* Example: API must be enabled on the Settings page in the product
+This plugin does not contain any requirements.
```
Comment (mhofert-r7) [inline diff/help.md]: see above

File: dirname/help.md [hunk 27]
```diff
@@ -14,14 +14,11 @@ This plugin is used to get the directory name of a file path or protocol and dom
...
-* Feature 1
-* Feature 2
-* Feature 3
+* Retrieve directory name of a file path or protocol and domain of a URL.
 
 # Requirements
 
-* Example: Requires an API Key from the product
-* Example: API must be enabled on the Settings page in the product
+This plugin does not contain any requirements.
```
Comment (mhofert-r7) [inline dirname/help.md]: see above

File: dirname/plugin.spec.yaml [hunk 28]
```diff
@@ -3,19 +3,20 @@ extension: plugin
...
-description: Dirname a path
+description: This plugin gets the directory name of a file path or protocol and domain of a URL
 version: 1.0.1
 vendor: rapid7
 support: community
 status: []
 resources:
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/dirname
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url:
```
Comment (mhofert-r7) [inline dirname/plugin.spec.yaml]: https://www.rapid7.com

File: freegeoip/help.md (hunk 16 above)
Comment (mhofert-r7) [inline freegeoip/help.md]: N/A plugin is obsolete

File: grep/plugin.spec.yaml [hunk 29]
```diff
@@ -3,20 +3,21 @@ extension: plugin
...
-description: Match pattern
+description: This plugin search for pattern in a string or a file
 version: 1.0.2
 vendor: rapid7
 support: community
 status: []
 resources:
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/grep
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
+  vendor_url:
```
Comment (mhofert-r7) [inline grep/plugin.spec.yaml]: https://www/rapid7.com

File: advanced_regex/help.md [hunk 30]
```diff
@@ -1,17 +1,15 @@
 # Description
 
-Perform regex operations on a string using Python regex.
+The Advanced Regex InsightConnect plugin perform regex operations on a string using Python regex.
```
Comment (jherlitz-r7) [inline advanced_regex/help.md]:
```suggestion
The Advanced Regex InsightConnect plugin performs regex operations on a string using Python regex.
```

File: advanced_regex/plugin.spec.yaml [hunk 31]
```diff
@@ -1,16 +1,22 @@
 plugin_spec_version: v2
+extension: plugin
+product: ["insightconnect"]
 name: advanced_regex
 title: Advanced Regex
-description: Perform advanced regex operations on a string using Python regex
+description: This plugin perform advanced regex operations on a string using Python regex
```
Comment (jherlitz-r7) [inline advanced_regex/plugin.spec.yaml]:
```suggestion
description: This plugin performs advanced regex operations on a string using Python regex
```

File: awk/plugin.spec.yaml [hunk 32]
```diff
@@ -3,20 +3,21 @@ extension: plugin
 product: ["insightconnect"]
 name: awk
 title: Awk
-description: Manipulate text with the GNU awk programming language
+description: This plugin manipulates input string or file with the GNU awk programming language
```
Comment (jherlitz-r7) [inline awk/plugin.spec.yaml]:
```suggestion
description: This plugin manipulates an input string or file with the GNU awk programming language
```

File: datetime/plugin.spec.yaml [hunk 33]
```diff
@@ -3,19 +3,20 @@ extension: plugin
 product: ["insightconnect"]
 name: datetime
 title: Datetime
-description: Actions for getting and manipulating timestamps
+description: This plugin manipulate timestamps using Python's maya library
```
Comment (jherlitz-r7) [inline datetime/plugin.spec.yaml]:
```suggestion
description: This plugin manipulates timestamps using Python's maya library
```

File: grep/plugin.spec.yaml [hunk 34]
```diff
@@ -3,20 +3,21 @@ extension: plugin
 product: ["insightconnect"]
 name: grep
 title: Grep
-description: Match pattern
+description: This plugin search for pattern in a string or a file
```
Comment (jherlitz-r7) [inline grep/plugin.spec.yaml]:
```suggestion
description: This plugin searches for a specified pattern in a string or a file
```

**Pull request merged**